pkg/controller/storagecluster/reconcile.go

```diff
@@ -917,11 +917,16 @@ func newStorageClassDeviceSets(sc *ocsv1.StorageCluster) []rook.StorageClassDevi
 	return storageClassDeviceSets
 }
 
-func (r *ReconcileStorageCluster) throttleStorageDevices(storageClassName string) (bool, error) {
+func (r *ReconcileStorageCluster) throttleStorageDevices(ds ocsv1.StorageDeviceSet) (bool, error) {
+	if ds.DataPVCTemplate.Spec.StorageClassName == nil || *ds.DataPVCTemplate.Spec.StorageClassName == "" {
+		return false, fmt.Errorf("no StorageClass specified")
+	}
+	storageClassName := *ds.DataPVCTemplate.Spec.StorageClassName
+
 	storageClass := &storagev1.StorageClass{}
 	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: "", Name: storageClassName}, storageClass)
 	if err != nil {
-		return false, fmt.Errorf("failed to retrieve Storage Class %q. %+v", storageClassName, err)
+		return false, fmt.Errorf("failed to retrieve StorageClass %q. %+v", storageClassName, err)
```

Comment:
Unrelated spelling change. Please exclude from this patch.

```suggestion
		return false, fmt.Errorf("failed to retrieve Storage Class %q. %+v", storageClassName, err)
```